src/Instrumentation/ReactHttp/tests/Integration/ReactHttpInstrumentationTest.php

```diff
@@ -0,0 +1,173 @@
+            ->withTracerProvider($this->tracerProvider)
+            ->withPropagator(TraceContextPropagator::getInstance())
+            ->activate();
+    }
+
+    public function tearDown(): void
+    {
+        $this->scope->detach();
+        $this->tracerProvider->shutdown();
+    }
+
+    public function test_fulfilled_promise(): void
+    {
+        $this->assertCount(0, $this->storage);
+
+        $promise = $this->browser->request('GET', 'http://example.com/success');
+        $promise->then();
+
+        $this->assertCount(1, $this->storage);
+
+        $span = $this->storage->offsetGet(0);
+        assert($span instanceof ImmutableSpan);
+        $this->assertSame('GET', $span->getAttributes()->get(TraceAttributes::HTTP_REQUEST_METHOD));
+        $this->assertSame('example.com', $span->getAttributes()->get(TraceAttributes::SERVER_ADDRESS));
+        $this->assertNull($span->getAttributes()->get(TraceAttributes::SERVER_PORT));
+        $this->assertSame('http://example.com/success', $span->getAttributes()->get(TraceAttributes::URL_FULL));
+        $this->assertSame('http', $span->getAttributes()->get(TraceAttributes::NETWORK_PROTOCOL_NAME));
+        $this->assertSame('1.1', $span->getAttributes()->get(TraceAttributes::NETWORK_PROTOCOL_VERSION));
+        $this->assertSame('', $span->getAttributes()->get(TraceAttributes::HTTP_REQUEST_BODY_SIZE));
+        $this->assertSame('http', $span->getAttributes()->get(TraceAttributes::URL_SCHEME));
+        $this->assertSame('ReactPHP/1', $span->getAttributes()->get(TraceAttributes::USER_AGENT_ORIGINAL));
+        $this->assertSame('React\Http\Io\Transaction::send', $span->getAttributes()->get(TraceAttributes::CODE_FUNCTION_NAME));
+        $this->assertStringEndsWith('vendor/react/http/src/Io/Transaction.php', $span->getAttributes()->get(TraceAttributes::CODE_FILE_PATH));
+        $this->assertSame(200, $span->getAttributes()->get(TraceAttributes::HTTP_RESPONSE_STATUS_CODE));
+        $this->assertSame('', $span->getAttributes()->get(TraceAttributes::HTTP_RESPONSE_BODY_SIZE));
+    }
+
+    /**
+     * Requires adding the following two lines to php.ini
+     * otel.instrumentation.http.request_headers[]="Accept"
+     * otel.instrumentation.http.response_headers[]="Content-Type"
+     */
+    /*public function test_tracked_headers(): void
+    {
+        $this->assertCount(0, $this->storage);
+
+        $promise = $this->browser->request('GET', 'http://example.com/success', ['Accept' => 'text/plain']);
+        $promise->then();
+
+        $this->assertCount(1, $this->storage);
+
+        $span = $this->storage->offsetGet(0);
+        assert($span instanceof ImmutableSpan);
+        $this->assertSame('text/plain', $span->getAttributes()->get(sprintf('%s.%s', TraceAttributes::HTTP_REQUEST_HEADER, 'accept')));
+        $this->assertSame('text/plain; charset=utf-8', $span->getAttributes()->get(sprintf('%s.%s', TraceAttributes::HTTP_RESPONSE_HEADER, 'content-type')));
+    }*/
```

Comment:
Is there anything missing here I can help out with?

Reply:
Nah, I just switched it to environment variables since those can be tested.